src/main/java/edu/berkeley/cs186/database/query/QueryPlan.java

```diff
@@ -683,19 +732,33 @@ private QueryOperator minCostOperator(Map<Set<String>, QueryOperator> map) {
      */
     public Iterator<Record> execute() {
         this.transaction.setAliasMap(this.aliases);
-        // TODO(proj3_part2): implement
-        // Pass 1: For each table, find the lowest cost QueryOperator to access
-        // the table. Construct a mapping of each table name to its lowest cost
-        // operator.
-        //
-        // Pass i: On each pass, use the results from the previous pass to find
-        // the lowest cost joins with each table from pass 1. Repeat until all
-        // tables have been joined.
-        //
-        // Set the final operator to the lowest cost operator from the last
-        // pass, add group by, project, sort and limit operators, and return an
-        // iterator over the final operator.
-        return this.executeNaive(); // TODO(proj3_part2): Replace this!
+        // TODO(proj3_part2): implement Done
+
+        // Pass 1: For each table, find the lowest cost QueryOperator to access the table.
+        Map<Set<String>, QueryOperator> pass1Map = new HashMap<>();
+        for (String table : this.tableNames) {
+            Set<String> singeTableSet = new HashSet<>();
+            singeTableSet.add(table);
```

Comment:
```suggestion
            Set<String> singeTableSet = new HashSet<>(Collections.singletonList(table));
```
pr에서 불편했던 점을 보고 제가 했던 방법 공유합니다! 이렇게 하면 간략화 할 수 있어 보기에 편하더라구요.

Reply:
여기도 singletonList를 쓸 수 있네요!!